ui/index.js

```diff
@@ -177,6 +181,23 @@ async function startApp(metamaskState, backgroundConnection, opts) {
     );
   }
 
+  // This block autoswitches chains based on the last chain used
+  // for a given dapp.  This allows the user to be connected on one chain
+  // for one dapp, and automatically change for another
```

Comment:
Nit: Maybe worth clarifying in this comment that this only switches if there are no pending confirmations? I had to dig into the `getAutomaticSwitchNetwork` to see that.